flatten-array.json

```diff
@@ -0,0 +1,19 @@
+{
+   "cases": [
+      {
```

Comment:
I wonder if there is a gentler build-up that we could use.

For example, what's the simplest possible thing? Perhaps `[]` -> `[]`. Then maybe `[[]]` -> `[]`.